internal/controller/capacity.go

```diff
@@ -191,10 +216,67 @@ func (*InstasliceReconciler) getStartIndexFromPreparedState(instaslice *inferenc
 		}
 
 	}
-
 	return newStart
 }
 
+func (r *InstasliceReconciler) actualProfileSliceSize(instaslice *inferencev1alpha1.Instaslice, profileName, gpuUUID string) int32 {
+	originalAllocatedIndex := gpuAllocatedSlices(instaslice, gpuUUID)
+	gpuAllocatedIndex := make([]int32, len(originalAllocatedIndex))
+	copy(gpuAllocatedIndex, originalAllocatedIndex) // Ensure we're not modifying the original state
+	actualSliceSize := int32(0)
+	startIdx := r.getStartIndexFromPreparedState(instaslice, profileName, gpuAllocatedIndex)
+	// Get required slice size
+	var neededContinousSlot int32
+	for profile, placement := range instaslice.Status.NodeResources.MigPlacement {
+		if profile == profileName {
+			neededContinousSlot = placement.Placements[0].Size
+			break
+		}
+	}
+	// Mark the allocated slots (simulate the placement)
+	for i := int32(0); i < neededContinousSlot; i++ {
+		if startIdx+i < int32(len(gpuAllocatedIndex)-1) {
+			gpuAllocatedIndex[startIdx+i] = 1
+			actualSliceSize++
+		}
+	}
+	return actualSliceSize
+}
+
+// getTotalFitForProfileOnGPU checks how many times a profile can fit on a given GPU.
+func (r *InstasliceReconciler) getTotalFitForProfileOnGPU(instaslice *inferencev1alpha1.Instaslice, profileName, gpuUUID string) int32 {
+	// Create a copy of the allocated GPU slots to simulate placements
+	originalAllocatedIndex := gpuAllocatedSlices(instaslice, gpuUUID)
+	gpuAllocatedIndex := make([]int32, len(originalAllocatedIndex))
+	copy(gpuAllocatedIndex, originalAllocatedIndex) // Ensure we're not modifying the original state
+	fitCount := int32(0)
+	for {
```

Comment:
running infinite for loop!

Reply:
The exit condition is when the index is 9, and getStartIndexFromPreparedState always returns 9 when the GPU is full but I have bounded the loop with the max GPU index to be extra safe.